pkg/api/mlflow/config/config_test.go

```diff
@@ -0,0 +1,83 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/rotisserie/eris"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestServiceConfig_Validate_Ok(t *testing.T) {
+	testData := []struct {
+		name           string
+		providedConfig *ServiceConfig
+		expectedConfig *ServiceConfig
+	}{
+		{
+			name: "ArtifactRootHasS3Prefix",
+			providedConfig: &ServiceConfig{
+				ArtifactRoot: "s3://bucket_name",
+			},
+			expectedConfig: &ServiceConfig{
+				ArtifactRoot: "s3://bucket_name",
+			},
+		},
+		{
+			name: "ArtifactRootHasFilePrefix",
+			providedConfig: &ServiceConfig{
+				ArtifactRoot: "file://path1/path2/path3",
+			},
+			expectedConfig: &ServiceConfig{
+				ArtifactRoot: "/pkg/api/mlflow/config/path1/path2/path3",
+			},
+		},
+		{
+			name: "ArtifactRootHasEmptyPrefix",
+			providedConfig: &ServiceConfig{
+				ArtifactRoot: "/path1/path2/path3",
+			},
+			expectedConfig: &ServiceConfig{
+				ArtifactRoot: "/path1/path2/path3",
+			},
+		},
+	}
```

Comment:
yes, it is already exists:
<img width="677" alt="Screenshot 2023-08-15 at 10 29 30" src="https://github.com/G-Research/fasttrackml/assets/7852635/25d8f10b-8e0a-404a-8b2c-feafcd86ef50">
and yes it will give `/path1/path2/path3`

Reply:
no, no, there are 2 cases: `file://...` and `file:///...` (notice the 3rd slash!)
`file://...` is providing a relative path whilst `file:///...` is providing an absolute path

you may want to change your unit test to use `assert.Equal` instead of `assert.Contains` and to compute the absolute path within the test too

Reply:
e.g. something like this:
```golang
func TestServiceConfig_Validate_Ok(t *testing.T) {
	mustAbs := func(path string) string {
		absPath, err := filepath.Abs(path)
		if err != nil {
			panic(err)
		}
		return absPath
	}

	testData := []struct {
		name           string
		providedConfig *ServiceConfig
		expectedConfig *ServiceConfig
	}{
		{
			name: "ArtifactRootHasS3Prefix",
			providedConfig: &ServiceConfig{
				ArtifactRoot: "s3://bucket_name",
			},
			expectedConfig: &ServiceConfig{
				ArtifactRoot: "s3://bucket_name",
			},
		},
		{
			name: "ArtifactRootHasFilePrefixAndIsRelative",
			providedConfig: &ServiceConfig{
				ArtifactRoot: "file://path1/path2/path3",
			},
			expectedConfig: &ServiceConfig{
				ArtifactRoot: mustAbs("path1/path2/path3"),
			},
		},
		{
			name: "ArtifactRootHasFilePrefixAndIsAbsolute",
			providedConfig: &ServiceConfig{
				ArtifactRoot: "file:///path1/path2/path3",
			},
			expectedConfig: &ServiceConfig{
				ArtifactRoot: "/path1/path2/path3",
			},
		},
		{
			name: "ArtifactRootHasEmptyPrefix",
			providedConfig: &ServiceConfig{
				ArtifactRoot: "/path1/path2/path3",
			},
			expectedConfig: &ServiceConfig{
				ArtifactRoot: "/path1/path2/path3",
			},
		},
	}

	for _, tt := range testData {
		t.Run(tt.name, func(t *testing.T) {
			assert.Nil(t, tt.providedConfig.Validate())
			assert.Equal(t, tt.providedConfig.ArtifactRoot, tt.expectedConfig.ArtifactRoot)
		})
	}
}
```

Reply:
ahh, I see now.

Reply:
done.